subprojects/launcher/src/main/java/org/gradle/launcher/cli/CommandLineActionFactory.java

```diff
@@ -69,6 +69,7 @@
  * <p>Responsible for converting a set of command-line arguments into a {@link Runnable} action.</p>
  */
 public class CommandLineActionFactory {
+    public static final String WELCOME_MESSAGE_ENABLED_SYSTEM_PROPERTY = "welcome.message.enabled";
```

Comment:
:x: To be in line with other such internal switches I would name this something like `org.gradle.internal.launcher.welcomMessageEnabled`

Reply:
done